app/src/main/java/org/simple/clinic/summary/PatientSummaryScreenController.kt

```diff
@@ -332,6 +333,16 @@ class PatientSummaryScreenController @Inject constructor(
         .mergeWith(showScheduleAppointmentSheetOnDoneClicks)
   }
 
+  private fun openLinkIdWithPatientSheet(events: Observable<UiEvent>): Observable<UiChange> {
+    return events
+        .ofType<PatientSummaryScreenCreated>()
+        .filter { it.openIntention is OpenIntention.LinkIdWithPatient }
```

Comment:
This can also be written as

```kotlin
events
  .ofType<PatientSummaryScreenCreated>()
  .map { it.openIntention }
  .ofType<LinkIdWithPatient>()
```

This has the advantage of not requiring to cast it again further down the chain.

Reply:
This is not feasible since I also need `it.patientUuid`.